forms/NumericField.php

```diff
@@ -2,13 +2,65 @@
 
 /**
  * Text input field with validation for numeric values. Supports validating
- * the numeric value as to the {@link i18n::get_locale()} value.
+ * the numeric value as to the {@link i18n::get_locale()} value, or an
+ * overridden locale specific to this field.
  * 
  * @package forms
  * @subpackage fields-formattedinput
  */
 class NumericField extends TextField {
 
+	/**
+	 * Override locale for this field
+	 * 
+	 * @var string
+	 */
+	protected $locale = null;
+
+	public function setValue($value, $data = array()) {
+		require_once "Zend/Locale/Format.php";
+
+		// If passing in a non-string number, or a value
+		// directly from a dataobject then localise this number
+		if ((is_numeric($value) && !is_string($value)) || 
+			($value && $data instanceof DataObject)
+		){
+			$locale = new Zend_Locale($this->getLocale());
+			$this->value = Zend_Locale_Format::toNumber($value, array('locale' => $locale));
+		} else {
+			// If an invalid number, store it anyway, but validate() will fail
+			$this->value = $this->clean($value);
+		}
+		return $this;
+	}
+
+	/**
+	 * In some cases and locales, validation expects non-breaking spaces
+	 *
+	 * @param string $input
+	 * @return string The input value, with all spaces replaced with non-breaking spaces
+	 */
+	protected function clean($input) {
+		$nbsp = html_entity_decode('&nbsp;', null, 'UTF-8');
+		return str_replace(' ', $nbsp, trim($input));
+	}
+	
+	/**
+	 * Determine if the current value is a valid number in the current locale
+	 * 
+	 * @param mixed $value Optional value to check against. Will check the current value if not given
+	 * @return bool
+	 */
+	public function IsNumeric($value = null) {
```

Comment:
Why start this function with an uppercase? We have this weird rule in the coding conventions about UpperCamelCase for anything that's callable from a template, but that pretty much applies to all public methods. Also, I think allowing to pass in a `$value` is making the API too confusing (using it as a quasi-static checker). What's the reasoning for this?